File: python/Program/Vehicle.py
#Saya Muhammad Yusdan Ali Batubara NIM 2206847 mengerjakan latihan 4 dalam mata 
#kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya maka saya 
#tidak melakukan kecurangan seperti yang telah dispesifikasikan Aamiin.

class Vehicle:
    def __init__(self, plat_nomor, merk, tahun_produksi, warna):
        self.plat_nomor = plat_nomor
        self.merk = merk
        self.tahun_produksi = tahun_produksi
        self.warna = warna

    #setter and getter
    
    def setPlatNomor(self, plat_nomor):
        self.plat_nomor = plat_nomor

    def getPlatNomor(self):
        return self.plat_nomor
    
    def setMerk(self, merk):
        self.merk = merk

    def getMerk(self):
        return self.merk
    
    def setTahunProduksi(self, tahun_produksi):
        self.tahun_produksi = tahun_produksi

    def getTahunProduksi(self):
        return self.tahun_produksi
    
    def setWarna(self, warna):
        self.warna = warna

    def getWarna(self):
        return self.warna

File: cpp/Vehicle.cpp
#include <iostream>
#include <string>
using namespace std;

class Vehicle {
private:
    string plat_nomor;
    string merk;
    int tahun_produksi;
    string warna;

public:
    // Constructor
    Vehicle(string plat_nomor, string merk, int tahun_produksi, string warna) {
        this->plat_nomor = plat_nomor;
        this->merk = merk;
        this->tahun_produksi = tahun_produksi;
        this->warna = warna;
    }

    // Setters
    void setPlatNomor(string plat_nomor) {
        this->plat_nomor = plat_nomor;
    }

    void setMerk(string merk) {
        this->merk = merk;
    }

    void setTahunProduksi(int tahun_produksi) {
        this->tahun_produksi = tahun_produksi;
    }

    void setWarna(string warna) {
        this->warna = warna;
    }

    // Getters
    string getPlatNomor() {
        return this->plat_nomor;
    }

    string getMerk() {
        return this->merk;
    }

    int getTahunProduksi() {
        return this->tahun_produksi;
    }

    string getWarna() {
        return this->warna;
    }
};
